mParticle-Singular/MPKitSingular.m

```diff
@@ -98,21 +108,18 @@ If your company SDK instance is available and is applicable (Please return nil i
     }
 }
 
-
 #pragma mark Application
 - (MPKitExecStatus *)checkForDeferredDeepLinkWithCompletionHandler:(void(^)(NSDictionary *linkInfo, NSError *error))completionHandler {
-    [Singular registerDeferredDeepLinkHandler:^(NSString *deeplink) {
-        NSDictionary *ddlLink = [[NSDictionary alloc] initWithObjectsAndKeys:deeplink,SINGULAR_DEEPLINK_KEY, nil];
+    if (_started && (ddlLink)){
```

Comment:
if neither of these are true, you should be saving a reference to the completion handler for later use.